handlers/authentication.go

```diff
@@ -1,72 +1,106 @@
 package handlers
 
 import (
+	"errors"
 	"fmt"
 	"net/http"
 	"time"
 
 	"k8s.io/client-go/tools/clientcmd/api"
 
-	"github.com/kiali/kiali/business/authentication"
 	"github.com/kiali/kiali/config"
+	"github.com/kiali/kiali/handlers/authentication"
 	"github.com/kiali/kiali/kubernetes"
 	"github.com/kiali/kiali/log"
 	"github.com/kiali/kiali/util/httputil"
 )
 
 type AuthenticationHandler struct {
-	conf                config.Config
+	conf                *config.Config
 	authController      authentication.AuthController
 	homeClusterSAClient kubernetes.ClientInterface
+	kialiSAClients      map[string]kubernetes.ClientInterface
+	authRedirectHandler http.Handler
 }
 
 type AuthInfo struct {
-	Strategy              string      `json:"strategy"`
-	AuthorizationEndpoint string      `json:"authorizationEndpoint,omitempty"`
-	LogoutEndpoint        string      `json:"logoutEndpoint,omitempty"`
-	LogoutRedirect        string      `json:"logoutRedirect,omitempty"`
-	SessionInfo           sessionInfo `json:"sessionInfo"`
-	SecretMissing         bool        `json:"secretMissing,omitempty"`
+	Strategy                        string            `json:"strategy"`
+	AuthorizationEndpoint           string            `json:"authorizationEndpoint,omitempty"`
```

Comment:
Is AuthorizationEndpoint still needed now that we have AuthorizationEndpointPerCluster? Perhaps a comment here to explain what this is now (I assume it is the endpoint for the "home cluster" but that's just a guess).

Reply:
I guess technically no but it's basically the auth endpoint for the cluster kiali is deployed on and one of the current requirements of this feature is that you login to that cluster so it's nice to have it separate as it requires less changes on the frontend that way.

Reply:
I would suggest adding a comment here so future devs reading this will understand. Otherwise, the name `AuthorizationEndpoint` doesn't explain this is the endpoint for the cluster that Kiali is deployed in. For example:

```suggestion
	// this is the endpoint for the cluster in which Kiali is deployed
	AuthorizationEndpoint           string            `json:"authorizationEndpoint,omitempty"`
```

Reply:
I'd still recommend this comment or something like it so people aren't confused and wonder what it is for